### PR TITLE
libcurl: Add ability to disable NTLM

### PR DESCRIPTION
**Pull request by exief**
Update the recipe for libcurl to allow for disabling NTLM

Internal developers using OpenSSL configurations that don't contain DES get build errors from curl with the current recipe as it builds in NTLM core which relies on DES being in OpenSSL.

This patch adds the ability for developers to disable NTLM and so that this portion does not get built.

---

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libcurl/all/conanfile.py [hunk 1]
```diff
@@ -64,6 +67,9 @@ class LibcurlConan(ConanFile):
         "with_c_ares": False,
         "with_proxy": True,
         "with_rtsp": True,
+        "with_crypto_auth": True,
+        "with_ntlm": True,
+        "with_ntlm_wb": True,
```
Comment (ruilvo) [inline recipes/libcurl/all/conanfile.py:72]: Do these defaults keep "backwards" compatibility with earlier revisions of the recipe?

Comment (exief) [reply recipes/libcurl/all/conanfile.py:72]: Yes it does 🙂

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — DISMISSED** [review]

**Comment by madebr** [comment]
Should libcurl add some checks in validate that, if `libcurl:with_ntlm` is True, `openssl:no_des` must be False?

**Review by prince-chrismc — APPROVED** [review]

**Review by prince-chrismc — CHANGES_REQUESTED** [review]
This is actually a good idea ! https://github.com/conan-io/conan-center-index/pull/6772#issuecomment-897791040

**Comment by exief** [comment]
> Should libcurl add some checks in `validate` that, if `libcurl:with_ntlm` is True, `openssl:no_des` must be False?

Sorry for delay in getting back to you, I'll add that in when I get a free moment

> This is actually a good idea ! [#6772 (comment)](https://github.com/conan-io/conan-center-index/pull/6772#issuecomment-897791040)

I agree definitely a good idea

File: recipes/libcurl/all/conanfile.py [hunk 2]
```diff
@@ -150,7 +150,7 @@ def configure(self):
         if self.options.with_ssl == "darwinssl" and not tools.is_apple_os(self.settings.os):
             raise ConanInvalidConfiguration("darwinssl only suppported on Apple like OS (Macos, iOS, watchOS or tvOS).")
         if self.options.with_ssl == "wolfssl" and self._is_using_cmake_build and tools.Version(self.version) < "7.70.0":
-            raise ConanInvalidConfiguration("Before 7.70.0, libcurl has no wolfssl support for Visual Studio or \"Windows to Android cross compilation\"")
+            raise ConanInvalidConfiguration("Before 7.70.0, libcurl has no wolfssl support for Visual Studio or \"Windows to Android cross compilation\"")
```
Comment (prince-chrismc) [inline recipes/libcurl/all/conanfile.py:153]:
```suggestion
            raise ConanInvalidConfiguration("Before 7.70.0, libcurl has no wolfssl support for Visual Studio or \"Windows to Android cross compilation\"")
```

File: recipes/libcurl/all/conanfile.py [hunk 3]
```diff
@@ -578,3 +578,8 @@ def package_info(self):
             self.cpp_info.components["curl"].requires.append("zstd::zstd")
         if self.options.with_c_ares:
             self.cpp_info.components["curl"].requires.append("c-ares::c-ares")
+
+    def validate(self):
+        if self.options.with_ssl == "openssl":
+            if self.options.with_ntlm and self.options["openssl"].no_des:
+                raise ConanInvalidConfiguration("Disabling DES with NTLM enabled not valid")
```
Comment (prince-chrismc) [inline recipes/libcurl/all/conanfile.py:585]:
```suggestion
                raise ConanInvalidConfiguration("option with_ntlm=True requires openssl:no_des=False")
```

**Comment by prince-chrismc** [comment]
Seems like the jobs cancelled in the wrong order

You'll need to retrigger CI, close the pr wait 10s and then re-open it 🔁

**Review by prince-chrismc — APPROVED** [review]

**Comment by exief** [comment]
> Seems like the jobs cancelled in the wrong order
> 
> You'll need to retrigger CI, close the pr wait 10s and then re-open it

looks like it auto-triggered while I was out! looks all green now :)

**Review by SSE4 — APPROVED** [review]

**Review by madebr — APPROVED** [review]

**Pull request merged**